Fix target team on request further clearance transitions

beccapearce opened the pull request.

misaka commented on config/routes.rb:
```diff
@@ -1,5 +1,7 @@
 Rails.application.routes.draw do
 
+  get 'heart_beat/ping'
```
This should probably be:

```ruby
get 'heartbeat/ping'
```

But, do we really need it since we have `/ping` routed below?

aliuk2012 replied:
You are right this shouldn't have been there. Fixed it

misaka commented on spec/controllers/heartbeat_controller_spec.rb:
```diff
@@ -0,0 +1,31 @@
+
+RSpec.describe HeartbeatController, type: :controller do
+
+  describe 'ping and heartbeat do not force ssl' do
```
Instead of stating the expectation here, we should be stating what we're testing ... I would expect to see the test in this one in the `describe '#ping'` below.

aliuk2012 replied:
Could you check this now?

EdwardAndress commented on spec/controllers/heartbeat_controller_spec.rb:
```diff
@@ -0,0 +1,31 @@
+RSpec.describe HeartbeatController, type: :controller do
+
+  describe '#ping' do
+    describe 'does not force ssl' do
```
Maybe I am being dumb but, I don't quite get this... how is a response code of 301 associated with SSL?

misaka replied:
What I was trying to suggest here is that `describe` block should stick to tell us the class/method/bit of code that's being speced, while the `it` block should specify specific behaviour. Something more akin to this:

```ruby
describe '#ping' do
  it 'does not force ssl' do
    allow(Rails).to receive(:env).and_return(double(development?: false, production?: true))
    get :ping
    expect(response.status).not_to eq(301)
  end

  it 'returns JSON with app information' do
...
```

The other thing, having thought about this for a tiny bit, is shouldn't this policy of redirecting HTTP to HTTPS be handled by NGinx? Is it actually be handled in the Rails app somewhere?

aliuk2012 replied:
Ah I see, good point there is no need to have a "before" statement for only one test.

I'm not sure about your NGINX question but I think that's a really good thought about where's the redirect happening.

aliuk2012 replied:
301 is suggesting there's a permanent redirect, i.e I use 'http://....' it then redirects to 'https://...'

aliuk2012 replied:
This got me thinking are we forcing ssl on both Ask and Answer Tool Apps? I'm not sure if this is being done at NGINX level but I've seen another service use [force_ssl](http://api.rubyonrails.org/classes/ActionController/ForceSSL/ClassMethods.html) in their application controller(https://github.com/ministryofjustice/advocate-defence-payments/blob/develop/app/controllers/application_controller.rb#L8)

Shouldn't we be doing something similar ?

aliuk2012 commented:
@EdwardAndress @misaka I think I've addressed all your concerns. Could you review?

aliuk2012 commented:
@EdwardAndress review/approval?

misaka committed:
CT-1523 re-add target user and team to further_clearance transitions

This was a bug introduced by CT-1467 work, where we assumed that cases wouldn't
need these values. In fact they are needed on the incoming cases page to display
the assigned responder as being the "requested by" team.

@becca

misaka committed:
Add db:seed:dev:teams task mention to readme

Commit:
CT-1523 adds tests for cases that have requested further clearance

Commit:
CT-1523 request further clearance feature test
